nixos/modules/installer/sd-card/sd-image.nix

```diff
@@ -150,6 +150,15 @@ in
         Whether to configure the sd image to expand it's partition on boot.
       '';
     };
+
+    extraRpiFirmwareConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra options that will be appended to the <literal>/boot/firmware/config.txt</literal> file.
+        For possible values, see: https://www.raspberrypi.org/documentation/configuration/config-txt/
+      '';
+    };
```

Comment:
Maybe you can use https://github.com/NixOS/nixpkgs/blob/d258316fba32587b63f1efa14df1f25a50fd25e3/pkgs/pkgs-lib/formats.nix#L60

that seems to be the preferred method after https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md

Reply:
Otherwise LGTM

Reply:
The config.txt is not quite ini, settings can have spaces and headers can be nested.

Reply:
I see, so honoring RFC42 might get slightly more involved, but neither unreasonably so as to have reasons to discard it.

https://github.com/NixOS/nixpkgs/blob/master/lib/generators.nix#L149

So `toGitINI` in there is an example how value generation can be customized relatively easily.

Since the arguments in RFC42 are quite strong and unambiguous, I think you surely might forgive me for seeking ways to comply, even at the burden of a little extra mile. :smile:

Reply:
The main difference from .ini is that the config file doesn't have sections, but [filters](https://www.raspberrypi.org/documentation/configuration/config-txt/conditional.md). I'm not sure entirely how to encode this in a structured way, since some filters combine and others change the semantics (the hdmi filters). Would a `listOf lines` that are merged with the resetting `[all]` filter interposed be acceptable to you?

I'd discourage removing this option since I need it personally, and I'm a don't-let-perfect-be-the-enemy-of-good type of person.

Reply:
`listOf lines` that sounds good to me. It provides at least some level of composability.

> I'm a don't-let-perfect-be-the-enemy-of-good type of person.

Especially so invoking the value of practical progress. :+1: 

I had to read up the filtering rules and indeed it seems a bit difficult. Since we are usually mapping settings to individual hosts in `nix`, this whole filtering is probably not necessary and `[all]` is a good catchall.